akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolMasterActor.scala

```diff
@@ -57,6 +60,11 @@ private[http] final class PoolMasterActor extends Actor with ActorLogging {
     context.watch(ref)
   }
 
+  private def createDeadline(gateway: PoolGateway): Option[Deadline] = gateway.hcps.setup.settings.idleTimeout match {
+    case finite: FiniteDuration ⇒ Some(finite.fromNow)
```

Comment:
Not sure if looking at the clock for each request might be expensive?

Reply:
Probably OK since I just read `Deadline` uses `System.nanoTime`, that should be nice and fast IIRC